SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/quote/view/QuoteConfirmationModal.kt

```diff
@@ -0,0 +1,533 @@
+import app.cybrid.sdkandroid.core.BigDecimal
+import app.cybrid.sdkandroid.core.BigDecimalPipe
+import app.cybrid.sdkandroid.core.toBigDecimal
+import app.cybrid.sdkandroid.ui.Theme.robotoFont
+
+private enum class QuoteConfirmationState {
+    PENDING, CONTENT, SUBMITTED, DONE
+}
+
+@Composable
+fun QuoteConfirmationModal(
+    viewModel: QuoteViewModel,
+    asset: MutableState<AssetBankModel>,
+    pairAsset: AssetBankModel,
+    showDialog: MutableState<Boolean>,
+    updateInterval: Long = 5000
+) {
+
+    // -- Modal State
+    val modalState:MutableState<QuoteConfirmationState> = remember { mutableStateOf(QuoteConfirmationState.PENDING) }
+
+    // -- Check state Pending -> Content
+    if (viewModel.quoteBankModel.guid != null) {
+        modalState.value = QuoteConfirmationState.CONTENT
+    }
+
+    // -- Check state Submitted -> Done
+    if (viewModel.tradeBankModel.guid != null) {
+        modalState.value = QuoteConfirmationState.DONE
+    }
+
+    // -- Content
+    Dialog(
+        onDismissRequest = {
+
+            viewModel.canUpdateQuote = true
+            showDialog.value = false
+            dismissModal(modalState, viewModel)
+        },
+        properties = DialogProperties(
+            usePlatformDefaultWidth = false
+        ),
+    ) {
+        Surface(
+            shape = RoundedCornerShape(28.dp),
+            color = colorResource(id = R.color.modal_color),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 20.dp, end = 20.dp)
+        ) {
+
+            when(modalState.value) {
+
+                QuoteConfirmationState.PENDING -> {
+                    QuoteConfirmationLoading(
+                        textID = R.string.trade_flow_quote_confirmation_modal_pending_title)
+                }
+
+                QuoteConfirmationState.CONTENT -> {
+                    QuoteConfirmationContent(
+                        viewModel = viewModel,
+                        asset = asset,
+                        pairAsset = pairAsset,
+                        showDialog = showDialog,
+                        modalState = modalState,
+                        refreshTime = updateInterval)
+                }
+
+                QuoteConfirmationState.SUBMITTED -> {
+                    QuoteConfirmationLoading(
+                        textID = R.string.trade_flow_quote_confirmation_modal_submitted_title)
+                }
+
+                QuoteConfirmationState.DONE -> {
+                    QuoteConfirmationContentDone(
+                        viewModel = viewModel,
+                        asset = asset,
+                        pairAsset = pairAsset,
+                        showDialog = showDialog,
+                        modalState = modalState
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun QuoteConfirmationLoading(textID:Int) {
+
+    Box(
+        modifier = Modifier
+            .height(320.dp)
```

Comment:
It's good to avoid magic numbers or constants spread all over the project and have them centralized for better maintenance. You could do that on a separate PR.

Reply:
No, that is a bad design desition, into Compose (@Composable functions) the context suggest use Strings, Colors, Images, but dimensions no, cause the new APi standard use "dp" and "sp" and if we use values files we need to handle "context" or "ContextResource" and cast or transform any dimension in Composable dimensions